### PR TITLE
Fix CloseIcon click

### LINKED ISSUE #58: [Login popup] The user closes Login popup by clicking outside of this popup
**Environment**: Windows10 21H2, Chrome 112.0.5615.138
**Reproducible**: always

**Preconditions**

The user enters Login popup window by clicking `Login button` in the upper right corner of the header

**Steps to reproduce**

Click outside of the Login popup

**Actual result**

Login popup window closes

**Expected result**

Nothing should happen

### LINKED ISSUE #115: [Login and Sign up pages] Window doesn't close
**Login and Sign up pages do not close, neither by clicking on the "x" nor by clicking outside the window, the only way to close the window is to click on "Join us for free"**

### PR DESCRIPTION
**Pull request by geersann**
###Issue
Open #115 Window doesn't close on button
Open #58 Window doesn't close outside 

###Description
Fix close modal window of CloseIcon. Add handleClose to close modal window on button and outside the box

###Assignees
assignees: @geersann

### CODE REVIEW

**Comment by geersann** [comment]
[https://github.com/WebUI-Project-Based-10-nr/Client/issues/115](url)

**Review by k7bart — APPROVED** [review]

**Review by petro97 — APPROVED** [review]

**Comment by Annaivv** [comment]
Now the modal closes when clicked on X icon. Should it also close by click on backdrop too?

**Review by dimaP2341 — APPROVED** [review]

File: src/components/popup-dialog/PopupDialog.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ const PopupDialog: FC<PopupDialogProps> = ({
         onMouseOver={handleMouseOver}
         sx={styles.box}
       >
-        <IconButton sx={styles.icon}>
+        <IconButton onClick={closeModalAfterDelay} sx={styles.icon}>
```
Comment (dimaP2341) [inline src/components/popup-dialog/PopupDialog.tsx:44]: Screenshot 2024-10-11 at 18 18 20

Comment (geersann) [reply src/components/popup-dialog/PopupDialog.tsx]: Already fix

**Comment by geersann** [comment]
> Now the modal closes when clicked on X icon. Should it also close by click on backdrop too?

I fixed it

**Review by hanus-kostiantyn — APPROVED** [review]

**Review by Yurii-04 — APPROVED** [review]

**Review by nrevutska — APPROVED** [review]

**Pull request merged**